### PR TITLE
Added update branch API

### LINKED ISSUE #1183: Support preview Update Branch API
GitHub Developer API announcement:
https://developer.github.com/changes/2019-05-29-update-branch-api/

This would be a great PR for any new contributor to this repo or a new Go developer.
All contributions are greatly appreciated!

Feel free to volunteer for any issue and the issue can be assigned to you so that others don't attempt to duplicate the work.

Please check out our [CONTRIBUTING.md](https://github.com/google/go-github/blob/master/CONTRIBUTING.md) guide to get started.

Thank you!

### PR DESCRIPTION
**Pull request by waseem18**
Fixes #1183

### CODE REVIEW

**Comment by waseem18** [comment]
@gmlewis Could you please review this PR?

File: github/pulls.go [hunk 1]
```diff
@@ -242,6 +242,49 @@ func (s *PullRequestsService) Create(ctx context.Context, owner string, repo str
 	return p, resp, nil
 }
 
+// PullReqestBranchUpdateOptions specifies the optional parameters to the
+// PullRequestsService.UpdateBranch method.
+type PullReqestBranchUpdateOptions struct {
+	// ExpectedHeadSha specifies the most recent commit on the pull request's branch.
+	// Default value is the SHA of the pull request's current HEAD ref.
+	ExpectedHeadSha *string `json:"expected_head_sha,omitempty"`
```
Comment (gmlewis) [inline github/pulls.go]: Let's please rename this to `ExpectedHeadSHA`.

File: github/pulls.go [hunk 2]
```diff
@@ -242,6 +242,49 @@ func (s *PullRequestsService) Create(ctx context.Context, owner string, repo str
...
+
+// PullRequestBranchUpdateResponse specifies the response of pull request branch update.
+type PullRequestBranchUpdateResponse struct {
+	Message *string `json:"message,omitempty"`
+	URL     *string `json:"url,omitempty"`
+}
+
+// UpdateBranch updates the pull request branch with latest upstream changes.
+//
+// This method might return an *AcceptedError and a status code of
```
Comment (gmlewis) [inline github/pulls.go]: Please remove the `*` for consistency.

**Comment by waseem18** [comment]
@gmlewis Addressed reviews. 

Hope the commits are squashed while merging.

**Comment by gmlewis** [comment]
Thank you, @vaibhavsingh97 for the LGTM.

Merging.

**Pull request merged**